kernel/fde/fde_test.go

```diff
@@ -528,6 +528,62 @@ func (s *fdeSuite) TestRevealErr(c *C) {
 	c.Check(osutil.FileExists(filepath.Join(dirs.GlobalRootDir, "/run/fde-reveal-key")), Equals, false)
 }
 
+func (s *fdeSuite) TestDeviceUnlock(c *C) {
+	checkSystemdRunOrSkip(c)
+
+	restore := fde.MockFdeInitramfsHelperCommandExtra([]string{"--user"})
+	defer restore()
+	fdeDeviceUnlockStdin := filepath.Join(c.MkDir(), "stdin")
+	mockSystemdRun := testutil.MockCommand(c, "fde-device-unlock", fmt.Sprintf(`
+cat - > %s
+printf "output-only-used-for-errors"
+`, fdeDeviceUnlockStdin))
+	defer mockSystemdRun.Restore()
+
+	mockKey := []byte("some-key")
+	p := fde.DeviceUnlockParams{
+		Key:    mockKey,
+		Device: "/dev/my-device",
+	}
+	err := fde.DeviceUnlock(&p)
+	c.Assert(err, IsNil)
+	c.Check(mockSystemdRun.Calls(), DeepEquals, [][]string{
+		{"fde-device-unlock"},
+	})
+	c.Check(fdeDeviceUnlockStdin, testutil.FileEquals, fmt.Sprintf(`{"op":"device-unlock","key":%q,"device":"/dev/my-device"}`, base64.StdEncoding.EncodeToString(mockKey)))
+
+	// ensure no tmp files are left behind
+	c.Check(osutil.FileExists(filepath.Join(dirs.GlobalRootDir, "/run/fde-device-unlock")), Equals, false)
+}
+
+func (s *fdeSuite) TestDeviceUnlockErr(c *C) {
+	checkSystemdRunOrSkip(c)
+
+	restore := fde.MockFdeInitramfsHelperCommandExtra([]string{"--user"})
+	defer restore()
+	mockSystemdRun := testutil.MockCommand(c, "fde-device-unlock", `
+echo  "output-only-used-for-errors" 1>&2
+sleep 0.2
```

Comment:
do we really need the sleep here for any reason?

Reply:
Thanks, I will remove it! IIRC its there because we had this issue that systemd-run would sometimes be racy